cmd/manager/operator.go

```diff
@@ -60,30 +60,29 @@ func init() {
 	defineOperatorFlags(operatorCmd)
 }
 
-func defineOperatorFlags(cmd *cobra.Command) {
-	cmd.Flags().Bool("skip-metrics", false,
-		"Skips adding metrics.")
-
-	// Add the zap logger flag set to the CLI. The flag set must
-	// be added before calling pflag.Parse().
-	flags := cmd.Flags()
-	flags.AddFlagSet(zap.FlagSet())
-
-	// Add flags registered by imported packages (e.g. glog and
-	// controller-runtime)
-	flags.AddGoFlagSet(flag.CommandLine)
-
-}
-
 // Change below variables to serve metrics on different host or port.
 var (
-	metricsHost               = "0.0.0.0"
-	metricsServiceName        = "metrics"
-	metricsPort         int32 = 8383
-	operatorMetricsPort int32 = 8686
-	defaultProducts           = []string{
-		"rhcos4",
-		"ocp4",
+	metricsHost                      = "0.0.0.0"
+	metricsServiceName               = "metrics"
+	metricsPort                int32 = 8383
+	operatorMetricsPort        int32 = 8686
+	defaultProductsPerPlatform       = map[PlatformType][]string{
+		PlatformOpenShift: {
+			"rhcos4",
+			"ocp4",
+		},
+		PlatformEKS: {
+			"eks",
+		},
+	}
+	defaultRolesPerPlatform = map[PlatformType][]string{
+		PlatformOpenShift: {
+			"master",
+			"worker",
+		},
+		"": {
```

Comment:
Some questions:
1. does a generic k8s distro not have master/worker or controlplane/worker roles by default?
2. wouldn't it be better to be explicit for EKS instead of relying on the `""` magic?

Reply:
1. No, they don't.
2. I expected `""` to be a default that could be used for other distros too. Seems Azure and GKE don't have those either.

Reply:
I am wondering for the second question, would `GenericPlatform` improve the readebility?

Reply:
@Vincent056 I like that idea! Let me do that.

Reply:
uhm... what if I re-use "UnknownPlatform" here?